### PR TITLE
Use structopt for command line arguments

### PR DESCRIPTION
**Pull request by nbaksalyar**
Using this library gives us more flexibility in ways users can configure the options provided by the library.
This PR also includes some changes in `directories`/config file loader handling (e.g., it also takes mobile platforms into account).

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use crate::error::Error;
 use serde::de::DeserializeOwned;
 use serde::Serialize;
 use std::fs::File;
-use std::io::{self, BufReader, BufWriter};
+use std::io::{BufReader, BufWriter};
```
Comment (ustulation) [inline src/utils.rs]: Why was this necessary

File: src/config.rs [hunk 2]
```diff
@@ -14,34 +14,48 @@ use crate::{NodeInfo, R};
...
     /// The interval is in milliseconds. A value of 0 disables this feature.
+    #[structopt(long)]
     pub idle_timeout_msec: Option<u64>,
     /// Interval to send keep-alives if we are idling so that the peer does not disconnect from us
     /// declaring us offline. If none is supplied we'll default to the documented constant.
     ///
     /// The interval is in milliseconds. A value of 0 disables this feature.
+    #[structopt(long)]
     pub keep_alive_interval_msec: Option<u32>,
     /// Path to our TLS Certificate. This file must contain `SerialisableCertificate` as content
```
Comment (ustulation) [inline src/config.rs]: Ah this is not correct - can we pls change it to say it's not a path we are asking but the certificate itself ?

File: src/utils.rs (hunk 1 above)
Comment (nbaksalyar) [reply src/utils.rs]: It's not used on non-T1/mobile platforms (see the `project_dir` impl below)
